### ISSUE TITLE
[Bug]: Videos do not play properly

### ISSUE THREAD

**Issue by CaptainMorgan12** [GeopJr/Tuba#749]
### Describe the bug

Environment: Debian 12, Tuba Flatpak install 0.6.1, GPU Vivante GC7000, etnaviv driver, gnome 3.38.5, ARM64, Librem 5 mobile.

### Steps To Reproduce

1. go to post with embedded video
2. select play video once loading completes
3. observe in e.g. 30sec video only 3-4 frames show

It does not play the video it just shows some downloaded frames based on slider position

### Logs and/or Screenshots

Not sure how to output logs for tuba.
![tuba](https://github.com/GeopJr/Tuba/assets/80182265/2e60dc58-4b09-4186-a7d0-f61eb4fbf9b9)

### Instance Backend

Mastodon

### Operating System

Debian 11

### Package

Flatpak

**Comment by gnuntoo** [comment]
I am having this problem on my librem 5 as well. I do wonder if this issue would fix itself on the next version of pureOS.

I was not getting any log output from playing the videos. That could possibly mean they're not relevant to the issue, but I'm not sure.

I'm going to try to gather more information, such as what kinds of videos it's having issues with. I can say that from playing several videos on my librem 5 with tuba, most had the problem stated in the post, aside from gifs and one other supposedly not-gif.

**Comment by GeopJr** [comment]
(sorry I just saw the edit)

> I see if i can downgrade to 0.2.0, 0.6.0 and see what happens. Do you have the Commit IDs for either? tried the log, but can't find them.

You can find the tag names and the commits on https://github.com/GeopJr/Tuba/releases, on the left side under the date. I'd recommend testing the latest `0.5` rather than way older versions as it'll be easier to point it down

> I do wonder if this issue would fix itself on the next version of pureOS.

Most likely not, if you are using the flatpak. I can't remember by heart but in a minor version bump I cleaned up the media viewer but that's pretty much it. Nothing really changed fundamentally on the way videos get loaded and passed to Gtk.Video. It *might* be a GTK related issue but unless someone makes a reproducer it might be difficult for them to pinpoint what the issue might be. (I do not have access to a linux mobile device so I can't really provide much help on this one)

**Comment by maltelau** [comment]
I'm experiencing a similar issue (could be unrelated since I'm on laptop rather than mobile).

Video preview image shows fine in timelines etc, but when I click it, it shows a spinner for a second, then goes to the black screen shown below: the media viewer doesn't show the video. Play button is unresponsive. Right click -> "Save Media" works fine, as does viewing the video in firefox. 

![Skærmbillede fra 2024-05-25 17-36-25](https://github.com/GeopJr/Tuba/assets/21195256/7319c3b2-250d-4bfe-8f1f-03d7a42f270e)

## Steps to reproduce
1. Find a post on mastodon with an embedded video, for example `https://todon.eu/@b9AcE/112502070070757575`
2. Click the video preview
Expected: plays video
Result: doesn't play video

## System
wm: i3
my instance: mastodon v4.3.0-nightly.2024-05-18+glitch
os: Debian GNU/Linux sid main non-free non-free-firmware contrib
prefix: /usr
flatpak: false
tuba version: 0.6.3 (production)
gtk: 4.12.5 (4.12.5)
libadwaita: 1.5.0 (1.5.0)
libsoup: 3.4.4 (3.4.4)
libgtksourceview: 5.12.0 (5.11.1)

**Comment by GeopJr** [comment]
Could you hover over the icon in the middle of the window (⛔) and see if it shows any errors? 

You are likely missing some codecs.

**Comment by maltelau** [comment]
Yes, hadn't noticed the tooltip. Loosely translated: `GTK couldn't find a media module. Check your installation.` Thanks for the quick response. Sorry for posting in what turns out to be an unrelated issue. 

This doesn't bring me much closer to solving it though. On one side: the video opens fine in vlc/mpv/firefox. On the other side: There is no stdout/stderr output, nothing in journalctl, and no context in the tooltip so I don't know what codecs "are missing" or where to look for that information. I'm guessing there is a gstreamer* package out there somewhere that should be added as a dependency?

**Comment by GeopJr** [comment]
I think the missing debian package is `libgtk-4-media-gstreamer`

Otherwise, these are the `gstreamer` packages I have installed on my debian sid vm, though you probably don't need most of them:

![image](https://github.com/GeopJr/Tuba/assets/18014039/aee4df78-b9f6-45c8-b691-bcfa25f38b17)

VLC, mpv and firefox work because they don't use gstreamer!

**Comment by maltelau** [comment]
👍 `libgtk-4-media-gstreamer` resolved it and should probably be added as a(n optional?) dependency. Thanks!

**Comment by GeopJr** [comment]
Probably fixed by: #1308
